src/material/tabs/tab-header.spec.ts

```diff
@@ -636,6 +636,23 @@ describe('MatTabHeader', () => {
       discardPeriodicTasks();
     }));
 
+    it('should re-align the ink bar when the size of a tab changes', fakeAsync(() => {
+      fixture = TestBed.createComponent(SimpleTabHeaderApp);
+      fixture.detectChanges();
+
+      const inkBar = fixture.componentInstance.tabHeader._inkBar;
+
+      spyOn(inkBar, 'alignToElement');
+
+      // Change font-size which would trigger a container width change
+      (fixture.elementRef.nativeElement as HTMLElement).style.fontSize = '20px';
```

Comment:
I suspect that this change is going to cause flakiness, because we can't assume that the resize observer will fire immediately.

Reply:
Could you explain? By definition, the ResizeObserver has to fire at the next frame before the paint and this is intentional and specced to avoid frame jank. I can deflake the test -- I just ran 50 times locally as well and never had a single flake.

Reply:
MDN has the following clarification:

> ResizeObserver avoids infinite callback loops and cyclic dependencies that are often created when resizing via a callback function. It does this by only processing elements deeper in the DOM in subsequent frames.

The way I'm reading this is that it's basically being invoked in a `requestAnimationFrame` which is asynchronous, whereas this test is synchronous.

Reply:
It's scheduled later but the change by definition has to be done before the paint is finished so that the change is reflected. I don't understand how the zone in the test is run to know if the change is reflected in the DOM but the test is not flakey on current runs. 

Did you have a preference on a way to test this that you would prefer? I'm open to suggestions. I noticed that other cases in the codebase that use RO don't even have tests in their PR that run as part of the changes, perhaps for a similar reason.